File: 202309/799.go
func champagneTower(poured int, query_row int, query_glass int) float64 {
	var dp [105][105]float64
	dp[0][0] = float64(poured)
	for i := 0; i < 100; i++ {
		for j := 0; j <= i; j++ {
			if dp[i][j] > 1 {
				half := (dp[i][j] - 1) / 2
				dp[i+1][j] += half
				dp[i+1][j+1] += half
			}
		}
	}
	if dp[query_row][query_glass] >= 1 {
		return 1
	}
	return dp[query_row][query_glass]
}

File: 202602/799.py
class Solution:
    def champagneTower(self, poured: int, query_row: int, query_glass: int) -> float:
        dp = [[0] * 101 for _ in range(101)]
        dp[0][0] = poured
        for i in range(query_row + 1):
            for j in range(query_row + 1):
                if dp[i][j] > 1:
                    half = (dp[i][j] - 1) / 2
                    dp[i + 1][j] += half
                    dp[i + 1][j + 1] += half
        return 1 if dp[query_row][query_glass] >= 1 else dp[query_row][query_glass]
